service/src/main/java/org/mskcc/cmo/metadb/service/impl/RequestServiceImpl.java

```diff
@@ -199,8 +199,14 @@ public Boolean requestHasMetadataUpdates(RequestMetadata existingRequestMetadata
             RequestMetadata requestMetadata) throws Exception {
         String existingMetadata = mapper.writeValueAsString(existingRequestMetadata);
         String currentMetadata = mapper.writeValueAsString(requestMetadata);
+        return requestHasMetadataUpdates(existingMetadata, currentMetadata);
+    }
+
+    @Override
+    public Boolean requestHasMetadataUpdates(String existingRequestMetadataJson,
+            String requestMetadataJson) throws Exception {
         try {
-            metadbJsonComparator.isConsistent(currentMetadata, existingMetadata);
+            metadbJsonComparator.isConsistent(requestMetadataJson, existingRequestMetadataJson);
```

Comment:
[requested change] Was this switch intentional? Which of these is considered the reference json and the target json?

Reply:
currentMetadata ->  requestMetadataJson
existingMetadata -> existingRequestMetadataJson

Reply:
definitely the existing request metadata, meaning the one that was fetched from the database, is the one that should be on the left

Reply:
this is fixed in the changes I am working on